dev-python/spyder/spyder-4.0.1.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit eutils distutils-r1 xdg-utils
+
+DESCRIPTION="The Scientific Python Development Environment"
+HOMEPAGE="
+	https://www.spyder-ide.org/
+	https://github.com/spyder-ide/spyder/
+	https://pypi.org/project/spyder/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+IUSE="hdf5 +webengine webkit"
+REQUIRED_USE="webengine? ( !webkit )"
+
+RDEPEND="
+	
+	dev-python/atomicwrites[${PYTHON_USEDEP}]
+	>=dev-python/chardet-2.0.0[${PYTHON_USEDEP}]
+	dev-python/cloudpickle[${PYTHON_USEDEP}]
+	dev-python/diff-match-patch[${PYTHON_USEDEP}]
+	dev-python/intervaltree[${PYTHON_USEDEP}]
+	dev-python/jedi[${PYTHON_USEDEP}]
+	dev-python/keyring[${PYTHON_USEDEP}]
+	dev-python/nbconvert[${PYTHON_USEDEP}]
+	dev-python/numpydoc[${PYTHON_USEDEP}]
+	dev-python/pexpect[${PYTHON_USEDEP}]
+	dev-python/pickleshare[${PYTHON_USEDEP}]
+	dev-python/psutil[${PYTHON_USEDEP}]
+	>=dev-python/pygments-2.0[${PYTHON_USEDEP}]
+	dev-python/pylint[${PYTHON_USEDEP}]
+	dev-python/pympler[${PYTHON_USEDEP}]
+	>=dev-python/python-language-server-0.31.2[${PYTHON_USEDEP}]
+	>=dev-python/pyxdg-0.26[${PYTHON_USEDEP}]
+	dev-python/pyzmq[${PYTHON_USEDEP}]
+	>=dev-python/qdarkstyle-2.7[${PYTHON_USEDEP}]
+	>=dev-python/qtawesome-0.5.7[${PYTHON_USEDEP}]
+	>=dev-python/qtconsole-4.6.0[${PYTHON_USEDEP}]
+	>=dev-python/qtconsole-4.5.0[${PYTHON_USEDEP}]
+	>=dev-python/QtPy-1.5.0[${PYTHON_USEDEP},svg,webengine?,webkit?]
+	>=sci-libs/Rtree-0.8.3[${PYTHON_USEDEP}]
+	dev-python/sphinx[${PYTHON_USEDEP}]
+	>=dev-python/spyder-kernels-1.8.1[${PYTHON_USEDEP}]
+	dev-python/watchdog[${PYTHON_USEDEP}]
+	hdf5? ( dev-python/h5py[${PYTHON_USEDEP}] )"
```

Comment:
`optfeature` is succesfully used in this ebuild, why wasn't this moved there? Why are webkit and webengine USE flags when they seemingly have no function in this ebuild, nor is either of them **forced**. 
https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Policies#USE-Controlled_Optional_RDEPENDs

Reply:
I did some digging into the source files, and you were correct. hdf5 is only required for one specific (internal) plugin, therefore I have moved it to optfeature. I also went ahead and added ebuilds for the other (external) plugins. (sorry I made this already big PR even bigger)

Regarding the tests on some of those packages, as with spyder and spyder-kernels etc, these tests are coverage tests (depend on pytest-cov) therefore I have not enabled them.